### PR TITLE
Dedupled search results DDG paths

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

## Which problem is this PR solving?
- Fixes https://github.com/jaegertracing/jaeger-ui/issues/503
 
## Short description of the changes
- De-duplicate DDG paths as suggested in the Issue #503

### CODE REVIEW

File: packages/jaeger-ui/src/model/ddg/transformTracesToPaths.tsx [hunk 1]
```diff
@@ -25,7 +25,9 @@ function transformTracesToPaths(
   focalService: string,
   focalOperation?: string
 ): TDdgPayload {
-  const dependencies: TDdgPayloadPath[] = [];
+  // const dependenciesMap: TDdgPayloadPath[] = [];
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/model/ddg/transformTracesToPaths.tsx:28]: remove this line?

Comment (rubenvp8510) [reply packages/jaeger-ui/src/model/ddg/transformTracesToPaths.tsx]: Ups,  removed, thanks.

File: packages/jaeger-ui/src/model/ddg/transformTracesToPaths.test.js [hunk 2]
```diff
@@ -164,11 +164,10 @@ describe('transform traces to ddg paths', () => {
...
-      new Set([
-        makeExpectedPath([rootSpan, focalSpan], clientTrace),
-        makeExpectedPath([rootSpan, focalSpan], kindlessTrace),
-      ])
-    );
+
+    const path = makeExpectedPath([rootSpan, focalSpan], clientTrace);
+    path.attributes.push({ key: 'exemplar_trace_id', value: kindlessTrace.data.traceID });
+
+    expect(new Set(result)).toEqual(new Set([path]));
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/model/ddg/transformTracesToPaths.test.js:171]: perhaps this should be in a separate test (with appropriate title about deduping), not in `span.kind === 'server'` test

Comment (rubenvp8510) [reply packages/jaeger-ui/src/model/ddg/transformTracesToPaths.test.js:171]: Yes, well I just fixed this test to pass, but I think this test wil be removed or replaces on this PR: https://github.com/jaegertracing/jaeger-ui/pull/557 as discussed here: https://github.com/jaegertracing/jaeger-ui/issues/554 WDYT?

Comment (yurishkuro) [reply packages/jaeger-ui/src/model/ddg/transformTracesToPaths.test.js:171]: even more important to have a standalone test for deduping.

**Pull request merged**